contract_manager/src/chains.ts

```diff
@@ -9,6 +19,25 @@ export abstract class Chain extends Storable {
   getId(): string {
     return this.id;
   }
+
+  /**
+   * Returns the payload for a governance SetFee instruction for contracts deployed on this chain
+   * @param fee the new fee to set
+   * @param exponent the new fee exponent to set
+   */
+  generateGovernanceSetFeePayload(fee: number, exponent: number): Buffer {
```

Comment:
I was also a bit on the fence on where to put these. Putting them in the contract classes gives a false sense that it can only be executed on that specific contract whereas it can be executed on any contract deployed on that chain (as long as the governance data source matches). Also, I'm planning to add a `GlobalChain` with id 0 for global messages.

Reply:
Fun fact, in aptos and sui, the target chain id is the same for testnet/mainnet!